widgets/src/qsharp_widgets/__init__.py

```diff
@@ -35,3 +36,46 @@ class EstimateDetails(anywidget.AnyWidget):
     def __init__(self, estimates):
         super().__init__()
         self.estimates = estimates
+
+
+class Histogram(anywidget.AnyWidget):
+    _esm = pathlib.Path(__file__).parent / "static" / "index.js"
+    _css = pathlib.Path(__file__).parent / "static" / "index.css"
+
+    comp = traitlets.Unicode("Histogram").tag(sync=True)
+    buckets = traitlets.Dict().tag(sync=True)
+    shot_count = traitlets.Integer().tag(sync=True)
+
+    _new_buckets = {}
+    _new_count = 0
+    _last_message = time.time()
+
+    results = []
+
+    def _update_ui(self):
```

Comment:
Is there a logic behind which names start with `_` here?

Reply:
It is internal, and not something intended for users  to call (see https://stackoverflow.com/a/1301369/1674945)

Reply:
I was familiar with the "internal" connotation, but I couldn't make sense of which ones are deemed internal here and why. What about `add_result`? And why are some parameters internal and some not? Maybe I'm just not getting it, but the internal-ness doesn't seem to line up with which ones actually get used outside this class and which ones don't.

Reply:
comp, buckets, and shot_count are part of the contract (those constructed from the 'traitlets' module are what get synced with the front-end for display). add_result should probably be internal, and result itself should probably only be access external via an API (if at all) so the user doesn't mutate it without some action. Will update for consistency.